packages/components/src/components/flyout/flyout.tsx

```diff
@@ -0,0 +1,229 @@
+      this.open,
+      this.position,
+      this.hasHeader,
+      this.hasFooter,
+      this.hasSubFooter,
+      this.theme
+    );
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+
+    const dismissBtn = (
+      <PrefixedTagNames.pButtonPure
+        class="dismiss"
+        type="button"
+        hideLabel
+        icon="close"
+        theme={this.theme}
+        onClick={this.dismissFlyout}
+        ref={(el) => (this.dismissBtn = el)}
+      >
+        Dismiss flyout
+      </PrefixedTagNames.pButtonPure>
+    );
+    return (
+      <Host onMouseDown={this.onMouseDown}>
+        <div
+          class="root"
+          role="dialog"
+          {...parseAndGetAriaAttributes({
+            'aria-modal': true,
+            'aria-hidden': !this.open,
+            ...parseAndGetAriaAttributes(this.aria),
+          })}
+          tabIndex={-1}
+          ref={(el) => (this.dialog = el)}
+          {...(this.hasSubFooter && { onScroll: this.onScroll })} // if no sub-footer is used scroll shadows are done via CSS
+        >
+          <div class="header" ref={(el) => (this.header = el)}>
+            {dismissBtn}
+            {this.hasHeader && (
+              <div class="header-content">
+                <slot name="header" />
+              </div>
+            )}
+          </div>
+          <div class="content">
+            <slot />
+          </div>
+          {this.hasFooter && (
+            <div class="footer" ref={(el) => (this.footer = el)}>
+              <slot name="footer" />
+            </div>
+          )}
+          {this.hasSubFooter && (
+            <div class="sub-footer" ref={(el) => (this.subFooter = el)}>
+              <slot name="sub-footer" />
+            </div>
+          )}
+        </div>
+      </Host>
+    );
+  }
+
+  private onMouseDown = (e: MouseEvent): void => {
+    if ((e.composedPath() as HTMLElement[])[0] === this.host && !clickStartedInScrollbarTrack(this.host, e)) {
+      this.dismissFlyout();
+    }
+  };
+
+  private updateFocusTrap = (isOpen: boolean): void => {
+    setFocusTrap(this.host, isOpen, this.dismissBtn, this.dismissFlyout);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  private onScroll = throttle(100, () => {
+    if (this.dialog.scrollHeight - this.dialog.clientHeight > 0) {
+      if (this.hasHeader) {
+        this.updateHeaderShadow();
+      }
+      if (this.hasFooter) {
+        this.updateFooterShadow();
+      }
+    }
+  });
+
+  private updateHeaderShadow = (): void => {
+    if (this.dialog.scrollTop > FLYOUT_SCROLL_SHADOW_THRESHOLD) {
+      this.header.classList.add('header--shadow');
+    } else {
+      this.header.classList.remove('header--shadow');
+    }
+  };
+
+  private updateFooterShadow = (): void => {
+    const shouldApplyShadow = this.subFooter.offsetTop > this.dialog.clientHeight + this.dialog.scrollTop;
+    if (shouldApplyShadow) {
+      this.footer.classList.add('footer--shadow');
```

Comment:
Could make `add` and `remove` the dynamic part since everything else is identical, e.g.
```
this.footer.classList[shouldApplyShadow ? 'add' : 'remove']('footer--shadow');
```